templates/resources/application_deployment.md.tmpl

```diff
@@ -0,0 +1,86 @@
+# {{.Name}} ({{.Type}})
```

Comment:
Don't know where to comment this, but can we add the new resources you've created to their examples usage
- Connector Applications
- Application Principal for Connector Application

Reply:
https://axualio.atlassian.net/browse/AXPD-8411

> “terraform apply with a newly defined application deployment resource results in a new connector running”
> terraform apply with the removal of a defined application deployment resource resulting in the connector being stopped and removed

So to answer your question:

> Creating application deployment starts the connector
> Updating application deployment stops the connector if it was running, updates the config, and starts it
> Deleting application deployment stops the connector if it was running, then deletes it

Reply:
Added a guide for Connector Application support here: docs/guides/connector-application.md

In addition to this, added documentation here:
- docs/resources/application_deployment.md
- docs/resources/application_principal.md
- docs/index.md